src/atomate2/qchem/schemas/task.py

```diff
@@ -0,0 +1,212 @@
+
+    final_energy: float = Field(
+        None, description="Final electronic energy for the calculation (units: Hartree)"
+    )
+    enthalpy: float = Field(
+        None, description="Total enthalpy of the molecule (units: kcal/mol)"
+    )
+    entropy: float = Field(
+        None, description="Total entropy of the molecule (units: cal/mol-K"
+    )
+
+    mulliken: List[Any] = Field(
+        None, description="Mulliken atomic partial charges and partial spins"
+    )
+    resp: List[float] = Field(
+        None,
+        description="Restrained Electrostatic Potential (RESP) atomic partial charges",
+    )
+    nbo: Dict[str, Any] = Field(
+        None, description="Natural Bonding Orbital (NBO) output"
+    )
+
+    frequencies: List[float] = Field(
+        None, description="Vibrational frequencies of the molecule (units: cm^-1)"
+    )
+
+    def as_dict(self) -> Dict[str, Any]:
+        return {
+            "@module": self.__class__.__module__,
+            "@class": self.__class__.__name__,
+            "initial_molecule": self.initial_molecule,
+            "optimized_molecule": self.optimized_molecule,
+            "final_energy": self.final_energy,
+            "enthalpy": self.enthalpy,
+            "entropy": self.entropy,
+            "mulliken": self.mulliken,
+            "resp": self.resp,
+            "nbo": self.nbo,
+            "frequencies": self.frequencies,
+        }
+
+
+class TaskDocument(BaseTaskDocument, MoleculeMetadata):
+    """
+    Definition of a Q-Chem task document
+    """
+
+    calc_code = "Q-Chem"
+    completed = True
+
+    is_valid: bool = Field(
+        True, description="Whether this task document passed validation or not"
+    )
+    state: QChemStatus = Field(None, description="State of this calculation")
+
+    cputime: float = Field(None, description="The system CPU time in seconds")
+    walltime: float = Field(None, description="The real elapsed time in seconds")
+
+    calcs_reversed: List[Dict] = Field(
+        [], description="The 'raw' calculation docs used to assembled this task"
+    )
+
+    orig: Dict[str, Any] = Field(
+        {}, description="Summary of the original Q-Chem inputs"
+    )
+    output = Field(OutputSummary())
+
+    critic2: Dict[str, Any] = Field(
+        None, description="Output from Critic2 critical point analysis code"
+    )
+    custom_smd: str = Field(
+        None, description="Parameter string for SMD implicit solvent model"
+    )
+
+    special_run_type: str = Field(
+        None, description="Special workflow name (if applicable)"
+    )
+
+    # TODO - type of `tags` field seems to differ among task databases
+    # sometimes List, sometimes Dict
+    # left as Any here to ensure tags don't cause validation to fail.
+    tags: Any = Field(None, description="Metadata tags")
+
+    warnings: Dict[str, bool] = Field(
+        None, description="Any warnings related to this task document"
+    )
+
```

Comment:
You will need to add a `from_directory` function to this task document/the one in emmet.